erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -666,6 +665,9 @@ def get_item_details(self, args=None, for_update=False):
 		if args.get("uom") and for_update:
 			ret.update(get_uom_details(args.get('item_code'), args.get('uom'), args.get('qty')))
 
+		if self.purpose == 'Material Issue':
+			ret["expense_account"] = frappe.get_cached_value('Company',  self.company,  "default_expense_account")
```

Comment:
It should be set based on `get_default_expense_account(args, item, item_group_defaults)`

Reply:
Not used get_default_expense_account because item.expense_account has found